src/app/frontend/deploy/deployfromfile_controller.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Controller for the deploy from file directive.
+ *
+ * @final
+ */
+export default class DeployFromFileController {
+  /** @ngInject */
+  constructor() {
+    /**
+     * It initializes the scope output parameter
+     *
+     * @export {!DeployFromFileController}
+     */
+    this.detail = this;
+
+    /**
+     * Unique id for file input element
+     *
+     * @export {string} */
+    this.fileInputId = `fileInput-${new Date().getTime()}`;
+
+    /**
+     * Custom file model for the selected file
+     *
+     * @export {{name:string, content:string}} */
+    this.file = {name: '', content: ''};
```

Comment:
Caveat: this will not work in serve:prod mode (please verify though). This is because this.file gets exported, but its props are renamed. Create two variables instead.